webapp/src/Controller/Jury/ContestController.php

```diff
@@ -504,8 +508,61 @@ public function editAction(Request $request, int $contestId)
                 $problem->setContest($contest);
             }
 
+            // Determine the removed teams, team categories and problems.
+            // Note that we do not send out create / update events for
+            // existing / new problems, teams and team categories. This happens
+            // when someone connects to the event feed (or we have a
+            // dependent event) anyway and adding the code here would
+            // overcomplicate this function
+            $deletedEntities = function(Collection $collection, string $idMethod) {
+                /** @var PersistentCollection $collection */
+                $deletedEntities = [];
+                foreach ($collection->getSnapshot() as $oldEntity) {
+                    $found = false;
+                    foreach ($collection->toArray() as $newEntity) {
+                        $newId = call_user_func([$newEntity, $idMethod]);
+                        $oldId = call_user_func([$oldEntity, $idMethod]);
+                        if ($newId === $oldId) {
+                            $found = true;
+                            break;
+                        }
+                    }
+
+                    if (!$found) {
+                        $deletedEntities[] = $oldEntity;
+                    }
+                }
+
+                return $deletedEntities;
+            };
+
+            /** @var Team[] $deletedTeams */
+            $deletedTeams = $deletedEntities($contest->getTeams(), 'getTeamid');
+            /** @var TeamCategory[] $deletedTeamCategories */
+            $deletedTeamCategories = $deletedEntities($contest->getTeamCategories(), 'getCategoryid');
+            /** @var ContestProblem[] $deletedProblems */
+            $deletedProblems = $deletedEntities($contest->getProblems(), 'getProbid');
+
             $this->saveEntity($this->em, $this->eventLogService, $this->dj, $contest,
                               $contest->getCid(), false);
+
+            $teamEndpoint         = $this->eventLogService->endpointForEntity(Team::class);
+            $teamCategoryEndpoint = $this->eventLogService->endpointForEntity(TeamCategory::class);
+            $problemEndpoint      = $this->eventLogService->endpointForEntity(Problem::class);
+
+            // TODO: cascade deletes. Maybe use getDependentEntities()?
+            foreach ($deletedTeams as $deletedTeam) {
```

Comment:
Nitpick: what about shorter `$team` instead of `$deletedTeam`? It's a tight for-loop and it's already clear that it's coming from an array `$deletedTeams`.